File: spring-data-requery/src/test/java/org/springframework/data/requery/domain/sample/AbstractAddress.java
package org.springframework.data.requery.domain.sample;

import io.requery.Embedded;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.requery.domain.AbstractComponent;

/**
 * AbstractAddress
 *
 * @author debop@coupang.com
 * @since 18. 6. 14
 */
@Getter
@Setter
@Embedded
public class AbstractAddress extends AbstractComponent {

    private static final long serialVersionUID = 6292061477905475831L;

    protected String country;
    protected String city;
    protected String streetName;
    protected String streetNo;

    public AbstractAddress() {}

//    public AbstractAddress(String country, String city, String streetName, String streetNo) {
//        this.country = country;
//        this.city = city;
//        this.streetName = streetName;
//        this.streetNo = streetNo;
//    }

}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/domain/sample/AbstractAddress.kt
package org.springframework.data.requery.domain.sample

import io.requery.Column
import io.requery.Embedded
import io.requery.Transient
import org.springframework.data.requery.domain.AbstractComponent
import org.springframework.data.requery.domain.ToStringBuilder

@Embedded
abstract class AbstractAddress: AbstractComponent() {

    @get:Column(length = 128)
    abstract var country: String?

    @get:Column(length = 64)
    abstract var city: String?

    @get:Column(length = 1024)
    abstract var streetName: String?

    @get:Column(length = 1024)
    abstract var streetNo: String?

    @Transient
    override fun buildStringHelper(): ToStringBuilder {
        return super.buildStringHelper()
            .add("country", country)
            .add("city", city)
            .add("streetName", streetName)
            .add("streetNo", streetNo)
    }
}